pkg/util/k8s/k8s.go

```diff
@@ -0,0 +1,591 @@
+package k8s
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateOrUpdateServiceAccount creates a service account if not present,
+// else updates it if it has changed
+func CreateOrUpdateServiceAccount(
+	k8sClient client.Client,
+	sa *v1.ServiceAccount,
+	ownerRef *metav1.OwnerReference,
+) error {
+	existingSA := &v1.ServiceAccount{}
+	err := k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      sa.Name,
+			Namespace: sa.Namespace,
+		},
+		existingSA,
+	)
+	if errors.IsNotFound(err) {
+		logrus.Debugf("Creating %s service account", sa.Name)
+		return k8sClient.Create(context.TODO(), sa)
+	} else if err != nil {
+		return err
+	}
+
+	modified := !reflect.DeepEqual(sa.ImagePullSecrets, existingSA.ImagePullSecrets)
+
+	for _, o := range existingSA.OwnerReferences {
+		if o.UID != ownerRef.UID {
```

Comment:
When would there be an owner reference which is not the operator?

Reply:
Generally it won't happen. Just in case someone deploys two StorageCluster objects, it will share the same service account. We won't know which storage cluster object they would delete. So adding owner reference for both.